middleware_test.go

```
@@ -0,0 +1,101 @@
+	client, err := New(server.URL[7:], &Options{
+		Creds:        credentials.NewStaticV4("mockkey", "mocksecret", ""),
+		Secure:       false,
+		Region:       "us-east-1",
+		Middlewares:  []Middleware{mw},
+	})
+	if err != nil {
+		t.Fatalf("Failed to create minio client: %v", err)
+	}
+
+	// Make a GetBucketPolicy request (which uses executeMethod under the hood)
+	_, err = client.GetBucketPolicy(context.Background(), "mybucket")
```

golangci-lint (2.12.2)

[error] ineffectual assignment to err

(ineffassign)